src/cluster/src/server.rs

```diff
@@ -0,0 +1,302 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! An interactive dataflow server.
+
+use std::fmt::{Debug, Formatter};
+use std::sync::{Arc, Mutex};
+
+use anyhow::{anyhow, Error};
+use async_trait::async_trait;
+use futures::future;
+use timely::communication::initialize::WorkerGuards;
+use timely::execute::execute_from;
+use timely::scheduling::SyncActivator;
+use timely::WorkerConfig;
+use tokio::runtime::Handle;
+use tokio::sync::mpsc;
+
+use mz_cluster_client::client::{ClusterStartupEpoch, TimelyConfig};
+use mz_ore::cast::CastFrom;
+use mz_ore::halt;
+use mz_ore::metrics::MetricsRegistry;
+use mz_persist_client::cache::PersistClientCache;
+use mz_service::client::{GenericClient, Partitionable, Partitioned};
+use mz_service::local::LocalClient;
+use tracing::{info, warn};
+
+use crate::communication::initialize_networking;
+
+type PartitionedClient<C, R> = Partitioned<LocalClient<C, R, SyncActivator>, C, R>;
+
+/// Configures a dataflow server.
+#[derive(Debug)]
+pub struct GenericConfig {
```

Comment:
Optional nit -- "Generic" to me usually implies Rust generics, which this struct isn't. Maybe another name like `SharedConfig` would be better? (And then a comment to say what it's shared between). Or `BaseConfig` ?

Reply:
Oh, _this_ is what you meant by generic! I was confused too. Recommend `ClusterConfig`.